### DISCUSSION TITLE
Action tag set value by expression

### DISCUSSION THREAD

**Discussion by TylerMiller-TomTom** [cloud-custodian/cloud-custodian#7359] [General]
When filtering for values properties of resources you're able to set the value type to expression to base the tested value against some value computed using JMESPath. I was wondering if there was a way to do the equivalent when setting the value of a tag. Specifically I need to be able to assign one tag based on the values of two other tags. I have not seen any documentation on such a feature or seen any signs of said feature when looking through the code. If this feature does not exist could this be considered a feature request. Integrating JMESPath doesn't seem that difficult from the code I saw in the filters, so unlike my previous question/request shouldn't require a major change. It doesn't make much sense to me to be able to select resources based on said filters, but not be able to automatically deal with them in a similar way.

**Comment by kapilt** [comment]
its come up before, its not supported at the moment for the aws provider. the reason underlying was that custodian tries to optimize for minimal api calls, for tagging we will preferentially use bulk apis where possible, where as this requires devolving to per resource api calls.  also our `tag:` syntax on filters isn't jmespath. so any contribution here would need to detect the use of object/resource expressions to fallback to per resource api calls.

**Comment by TylerMiller-TomTom** [comment]
I don't get what you mean by "the tag syntax on filters isn't jmespath" I've looked at them through the code, and I use a jmespath query in my filter. type value, value_type set to expression, and then the value property set to a jmespath expression. Like so:
```
- type: value
  key: "tag:name"
  op: eq
  value: "join('-', [type,tags.purpose])"
  value_type: expr
```
Anyways, without the ability to dynamically tag resources the usefulness of this software to automatically correct issues with tagging rather than just reporting them, is severely limited. Also we are using Azure, although I don't expect that changes anything with the availability of such a feature.
